src/edwards.rs

```diff
@@ -556,12 +556,11 @@ impl MultiscalarMul for EdwardsPoint {
 impl VartimeMultiscalarMul for EdwardsPoint {
     type Point = EdwardsPoint;
     
-    fn vartime_multiscalar_mul<I, J>(scalars: I, points: J) -> EdwardsPoint
+    fn optional_multiscalar_mul<I, J>(scalars: I, points: J) -> Option<EdwardsPoint>
```

Comment:
What is the reasoning for eliminating the `vartime_multiscalar_mul` method entirely? What if the user has already validated the points?